src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -43,6 +43,12 @@ public static partial class JsonSerializer
                         Debug.Assert(state.Current.JsonClassInfo != default);
 
                         ReadOnlySpan<byte> propertyName = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+                        if (reader._stringHasEscaping)
```

Comment:
This whole creation of `ReadOnlySpan<byte> propertyName` should happen inside the if block that the span is actually being used.

Move to inside here:
https://github.com/dotnet/corefx/blob/be77af0a58a0ea8399f93d86d3338f10095df104/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs#L64-L73

Reply:
Thanks